Allow logging for API requests

tumtubs opened the pull request:
LoggingVerbosity sets verbosity level.
GraphAPI allows logger to be supplied
Logging added to facilitate debugging errors from Facebook Graph API

martey commented on facebook/__init__.py:
```diff
@@ -31,6 +31,7 @@
 import requests
 import json
 import re
+from enum import Enum
```
This import isn't Python 2.7 compatible.

tumtubs replied:
Fixed in https://github.com/mobolic/facebook-sdk/pull/463/commits/b11fe55561d078fd3e3f91b9b09c92fc4f99e611

martey commented on facebook/__init__.py:
```diff
@@ -298,18 +313,39 @@ def _add_to_post_args_or_args(arg_name, arg_value):
+                self.logger.debug('Request method: %s' % method)
+                self.logger.debug('Request URL: %s' % url)
+                self.logger.debug('Request Args: %s' % str(args))
+                self.logger.debug('Request Post Args: %s' % str(post_args))
```
You should use `.format()` for formatting strings instead of `%`.

tumtubs replied:
`.format()` added in https://github.com/mobolic/facebook-sdk/pull/463/commits/b11fe55561d078fd3e3f91b9b09c92fc4f99e611

martey commented:
We use `black -l 79` to ensure that lines are under 80 characters. Sorry, the contributing guidelines should be clearer on this point.

martey commented:
@tumtubs is there a reason why this keeps being closed and reopened?

tumtubs commented:
> @tumtubs is there a reason why this keeps being closed and reopened?

@martey Just wanted to prompt that this PR is ready for further review. The previous comments have been addressed. Are there any other outstanding issues?

martey commented:
@tumtubs You already sent me a request to review this pull request that I received. I'll review this when I have the bandwidth to do so.

Closing and reopening this repeatedly isn't going to free up my schedule or get this merged faster. Unlike writing a comment or sending me an email, it doesn't even tell me that you would like an update.

Unless the contributing guidelines on a repository specifically ask you to close and reopen a pull request to attract a maintainer's attention, I would advise you never to do so. This is especially true if differing timezones mean that your closing/reopening will occur when the maintainer is asleep, that you close/reopen on two successive nights, and that the emails that your actions generate wake the maintainer up both times.

tumtubs commented:
> Your linting changes are modifying unrelated files and areas of the codebase - including files that do not need to be linted.
> 
> I would also appreciate it if you could compress all of your changes into one single commit.

@martey I've reverted the linting for non-relevant files and squashed the commits: https://github.com/mobolic/facebook-sdk/pull/463/commits/24da88d81979cfdc3f5397e2c5a7d7f4fd6ffb04

martey commented on facebook/__init__.py:
```diff
@@ -96,6 +143,8 @@ def __init__(
         proxies=None,
         session=None,
         app_secret=None,
+        logger=None,  # expected to contain logging level's debug() and error()
```
This comment doesn't make it clear what this parameter is supposed to contain. The "API Reference" in the documentation should also be updated when new parameters are added.

martey commented:
Closing this pull request since it has been almost a year since changes were requested, and the most recent changes make this pull request *less* mergeable than before (this library, along with most other Python libraries, has stopped supporting Python 2.7).
